applications/serial_lte_modem/src/nrfcloud/slm_at_nrfcloud.h

```diff
@@ -7,11 +7,31 @@
 #ifndef SLM_AT_NRFCLOUD_
 #define SLM_AT_NRFCLOUD_
 
-/**@file slm_at_nrfcloud.h
+/** @file slm_at_nrfcloud.h
  *
  * @brief Vendor-specific AT command for nRF Cloud service.
  * @{
  */
+#if defined(CONFIG_SLM_NRF_CLOUD)
+
+#include <stdbool.h>
+#include <modem/at_cmd_parser.h>
+
+/* Whether the connection to nRF Cloud is ready. */
+extern bool g_nrf_cloud_ready;
```

Comment:
I think that "extern" is used most commonly in C source instead of header.

Reply:
https://en.cppreference.com/w/c/language/storage_duration

Usage of `extern` in C file is a sign of bad code design and a source of bugs that compiler cannot detect. You might accidentally violate [DCL40-C. Do not create incompatible declarations of the same function or object](https://wiki.sei.cmu.edu/confluence/display/c/DCL40-C.+Do+not+create+incompatible+declarations+of+the+same+function+or+object).

So if a variable must be shared between C files, this is the correct way. Declare it in the header as `extern`. Define it once in C file.

Reply:
Thanks for the knowledge